javaslang/src/main/java/javaslang/collection/BitMappedTrie.java

```diff
@@ -55,6 +56,24 @@ private static int treeSize(int branchCount, int depthShift) {
         return branchCount * fullBranchSize;
     }
 
+    static <T> BitMappedTrie<T> ofAll(Iterable<? extends T> iterable, int size) {
+        if (size == 0) { return empty(); }
+
+        Object[] array = (iterable instanceof Collection<?>) ? ((Collection<?>) iterable).toArray()
+                                                             : asArray(iterable.iterator(), size);
+        return ofAll(array, size, 0);
+    }
+
+    static <T> BitMappedTrie<T> ofAll(Object[] array, int size, int depthShift) {
+        if (size == 0) { return empty(); }
+
+        int shift = depthShift;
+        for (; array.length > branchingFactor(); shift += BRANCHING_BASE) {
```

Comment:
Why do you think a constant method is weakening?
I need `BRANCHING_BASE` to be changeable because it's very difficult to test and debug a tree that huge.

Also, constants could be inlined in other places (e.g. maybe a `2 * BRANCHING_FACTOR` would be inlined) and changing the final variable from the test wouldn't change this particular usage.

I would prefer keeping it a method, or making the `BRANCHING_BASE` for the tests also (would only be able to test 2-3 levels that way)